src/main/java/thebetweenlands/common/item/tools/ItemBLBucketEmpty.java

```diff
@@ -111,8 +114,40 @@ public ItemBLBucketEmpty() {
         }
     }
 
-    private ItemStack fillBucket(ItemStack emptyBuckets, EntityPlayer player, Fluid fluid) {
-        ItemStack fullBucket = UniversalBucket.getFilledBucket(this.getFilledBucket(), fluid);
+    public static ItemStack getFilledBucket(@Nonnull UniversalBucket item, FluidStack fluidStack) {
+        Fluid fluid = fluidStack.getFluid();
+
+        if (fluidStack.tag == null || fluidStack.tag.hasNoTags())
+        {
+            if (fluid == net.minecraftforge.fluids.FluidRegistry.WATER)
+            {
+                return new ItemStack(Items.WATER_BUCKET);
```

Comment:
Are you sure this is right? I don't think so.

Reply:
In what way do you think this is incorrect, this what the regular getFilledBucket metod does and what it did in 1.10 as well, this is just modified to take a custom bucket item instead of using the default forge dynamic bucket when using the fill method.

Reply:
This returns a vanilla water bucket, inconsistent with the usage of wooden buckets

Reply:
This would never happen anyway as the buckets can't pickup water or lava. 
I guess because of that and that the specific usage of this method is for these buckets it could be removed.